src/clj/fluree/db/query/exec/eval.cljc

```diff
@@ -289,21 +296,35 @@
   [term expressions]
   (contains? (set expressions) term))
 
+(def context-var
+  (symbol "$-CONTEXT"))
+
+(defmacro iri
+  [s]
+  `(-> ~s
+       (json-ld/expand-iri ~context-var)
```

Comment:
Just trying to understand this - why wouldn't an IRI already be expanded by the time it gets here?

Reply:
Because that's what this query "function" is doing. The user puts this inside of a query function definition. The parser isn't going to be able to expand iris during parse time because it might not have access to enough information for the expansion. It has to be expanded when the function is evaluated. For example:

```clojure
[:filter (= ?s (iri (str ?userNs ":" ?firstName)))]
```